src/main/java/com/shopping_app/shoppingApp/service/CartService.java

```diff
@@ -0,0 +1,115 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.BaseException;
+import com.shopping_app.shoppingApp.Exceptions.NotFoundException;
+import com.shopping_app.shoppingApp.domain.Cart;
+import com.shopping_app.shoppingApp.domain.CartItems;
+import com.shopping_app.shoppingApp.domain.Product;
+import com.shopping_app.shoppingApp.model.Cart.CartAddRequest;
+import com.shopping_app.shoppingApp.model.Cart.CartItemUpdateRequest;
+import com.shopping_app.shoppingApp.model.Cart.CartItemResponse;
+import com.shopping_app.shoppingApp.model.Cart.CartResponse;
+import com.shopping_app.shoppingApp.repository.CartItemRepository;
+import com.shopping_app.shoppingApp.repository.CartRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Transactional
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class CartService {
+
+    private final CartRepository cartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    private final ProductService productService;
+
+    public CartResponse getUserCart(Long userId) {
+        Cart userCart = getCartByUserId(userId);
+        Set<CartItemResponse> cartItemResponses = userCart.getCartItems().stream().map(CartItemResponse::from).collect(Collectors.toSet());
+        return CartResponse.from(userCart, cartItemResponses);
+    }
+
+    public CartResponse addItemsToCart(CartAddRequest cartRequest, Long userId) {
+        Cart userCart = getCartByUserId(userId);
+        Product product = productService.findProductById(cartRequest.getProduct_id());
+        Optional<CartItems> cartItemProduct = cartItemRepository.findByCartIdAndProductId(userCart.getId(), cartRequest.getProduct_id());
+        checkAvailableQuantityOfProduct(product, cartRequest.getQuantity());
+        CartItems cartItem = new CartItems();
+        if (cartItemProduct.isEmpty()) {
+            cartItem.setProduct(product);
+            cartItem.setQuantity(cartRequest.getQuantity());
+            cartItem.setCart(userCart);
+            userCart.getCartItems().add(cartItem);
+        } else {
+            increaseQuantityInExistingItem(userCart, cartRequest.getProduct_id(), cartRequest.getQuantity());
```

Comment:
```suggestion
           updateCartItem(userCart, cartRequest.getProduct_id(), cartRequest.getQuantity());
```
see if you reuse the code for updating the cart item, since the same thing is done here and for the update cart item as well

Reply:
Yes , reused the code